sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/admin/PeriodicProvisioningTask.kt

```diff
@@ -72,19 +68,27 @@ class PreallocateProfilesTask(
                         }
                     }
 
-    private fun batchPreprovisionSimProfiles(hlrEntry: HssEntry,
-                                             profile: String,
+    private fun batchPreprovisionSimProfiles(hssEntry: HssEntry,
+                                             simProfileName: String,
                                              profileStats: SimProfileKeyStatistics) {
         val noOfProfilesToActuallyAllocate =
                 Math.min(maxNoOfProfileToAllocate.toLong(), profileStats.noOfUnallocatedEntries)
 
         for (i in 1..noOfProfilesToActuallyAllocate) {
-            simInventoryDAO.findNextNonProvisionedSimProfileForHss(hssId = hlrEntry.id, profile = profile)
+
+            // XXX This is all well, if allocation doesn't fail, but if it fails, it will try the
+            //     same profile forever, so something else should be done e.g. setting the
+            //     state of the profile to "provision failed" or something of that nature, so that it
+            //     is possible to move along.   This is an error in the logic of this code.
+            simInventoryDAO.findNextNonProvisionedSimProfileForHss(hssId = hssEntry.id, profile = simProfileName)
                     .flatMap { simEntry ->
-                        preProvisionSimProfile(hlrEntry, simEntry)
+                        preProvisionSimProfile(hssEntry, simEntry)
                                 .mapLeft {
                                     logger.error("Preallocation of SIM ICCID {} failed with error: {}}",
                                             simEntry.iccid, it.description)
+
+                                    // XXX This looks weird, get review on it (and the code above)
+                                    simInventoryDAO.setSmDpPlusState(simEntry.id!!, SmDpPlusState.ALLOCATION_FAILED)
```

Comment:
Would perfer that "provision-state" is used for marking invalid entires. That is extend the `ProvisionState` enum in `SimInventoryDAO` with the "ALLOCATION_FAILED" field and use the `dao.setProvisionState()` call. In this way we will have only "official" SmDpPlus states. Plus, keeping the SmDpPlus state unchanged will tell us at which state the entry was in when the allocation for some reason failed.

Reply:
ok